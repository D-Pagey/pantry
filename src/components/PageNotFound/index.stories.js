import React from 'react';
import { storiesOf } from '@storybook/react';
import PageNotFound from '.';

storiesOf('PageNotFound', module).add('default', () => <PageNotFound />);
