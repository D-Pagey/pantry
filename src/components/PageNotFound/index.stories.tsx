import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { PageNotFound } from '.';

export default { title: 'PageNotFound' };

export const normal = (): JSX.Element => (
    <MemoryRouter>
        <PageNotFound />
    </MemoryRouter>
);
