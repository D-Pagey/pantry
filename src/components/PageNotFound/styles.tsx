import styled from 'styled-components';
import { Link } from 'react-router-dom';

export const Title = styled.h1`
    color: blue;
    text-align: center;
`;

export const RouterLink = styled(Link)`
    color: red;
    text-align: center;
    display: block;
    font-size: 22px;
`;
