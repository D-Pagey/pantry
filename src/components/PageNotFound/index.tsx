import React from 'react';
import * as S from './styles';

export const PageNotFound = (): JSX.Element => (
    <div>
        <S.Title>404 - This page does not exist</S.Title>
        <S.RouterLink to="/">Return Home</S.RouterLink>
    </div>
);
