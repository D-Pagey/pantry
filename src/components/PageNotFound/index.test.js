import React from 'react';
import PageNotFound from '.';

describe('PageNotFound component', () => {
    it('should render', () => {
        const { container } = render(<PageNotFound />);
        expect(container.firstChild).toMatchSnapshot();
    });
});
