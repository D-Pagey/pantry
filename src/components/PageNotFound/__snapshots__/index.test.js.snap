// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNotFound component should render 1`] = `
.c0 {
  color: blue;
  text-align: center;
}

.c1 {
  color: red;
  text-align: center;
  display: block;
  font-size: 22px;
}

<div>
  <h1
    class="c0"
  >
    404 - This page does not exist
  </h1>
  <a
    class="c1"
    href="/"
  >
    Return Home
  </a>
</div>
`;
